### PR TITLE
Moo 36

### PR DESCRIPTION
**Pull request by bodavivi**

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/flowmoonstone/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+  @Autowired
+  UserService userService;
+
+  @GetMapping
+  public String getUser() {
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/controllers/UserController.java]: do we need this endpoint? :)

File: src/main/java/com/greenfoxacademy/flowmoonstone/models/UserDto.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package com.greenfoxacademy.flowmoonstone.models;
+
+import java.io.Serializable;
+
+public class UserDto implements Serializable {
+  private static final long serialVersionUID = -5636285383751642930L;
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/models/UserDto.java]: do we need this field?

File: src/main/java/com/greenfoxacademy/flowmoonstone/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+  @Autowired
+  UserService userService;
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/controllers/UserController.java]: dependency injection should be done through constructor, not through the field

File: src/main/java/com/greenfoxacademy/flowmoonstone/models/UserEntity.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "users")
+public class UserEntity implements Serializable {
+  private static final long serialVersionUID = -346854070220930471L;
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/models/UserEntity.java]: why do we have this field?

File: src/main/java/com/greenfoxacademy/flowmoonstone/security/SecurityConstants.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.greenfoxacademy.flowmoonstone.security;
+
+public class SecurityConstants {
+  public static final long EXPIRATION_TIME = 864000000;
+  public static final String TOKEN_PREFIX = "Bearer ";
+  public static final String HEADER_STRING = "Authorization";
+  public static final String SIGN_UP_URL = "/users/registration";
+  public static final String TOKEN_SECRET = "lsgs";
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/security/SecurityConstants.java]: this should be an environment variable

File: src/main/java/com/greenfoxacademy/flowmoonstone/services/user/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+  @Autowired
+  UserRepository userRepository;
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/flowmoonstone/services/user/UserServiceImpl.java]: dependency injection should be done through constructor

File: src/main/resources/schema.sql [hunk 7]
```diff
@@ -30,4 +31,16 @@ CREATE TABLE riverside_place
...
+(
+    id       int primary key auto_increment,
+    user_id  varchar(255),
+    first_name  varchar(50),
+    last_name   varchar(50),
+    email    varchar(120),
+    encrypted_password  varchar(255),
+    email_verification_token    varchar(255),
+    email_verification_status   bit(1)
 );
```
Comment (bekobarna) [inline src/main/resources/schema.sql]: empty line at the end of the file

File: src/main/java/com/greenfoxacademy/flowmoonstone/models/UserDto.java (hunk 2 above)
Comment (bodavivi) [reply src/main/java/com/greenfoxacademy/flowmoonstone/models/UserDto.java]: I read in several article that serialVersionUID is highly recommanded
should i delete it anyway?

**Pull request merged**